### PR TITLE
Optimize TSQL outbox cleanup by reducing batch size and adding rowlock hint

### PR DESCRIPTION
**Pull request by ramonsmits**
Optimize TSQL outbox cleanup by

1. Reducing batch size to prevent lock escalation to page/table locks 
2. Add hint to prevent escalation from row to page locks to not conflict with regular incoming message outbox operations as DELETE does not have to be efficient and should avoid any lock escalation over performance

To prevent lock escalation to page/table locks the number of rows must not be over 4.000 or 5.000 rows

Google states multiple resources that either going over 4.000 or 5.000 rows results into lock escalation from row to page/table locks:

- https://dba.stackexchange.com/questions/284675/delete-rows-older-than-x-days-without-locking-table
- https://www.sqlservercentral.com/forums/topic/row-locks-not-escalating-to-table-locks-after-5000
- https://www.sqlpassion.at/archive/2016/05/09/lock-escalations-do-they-always-happen/

### CODE REVIEW

**Review by andreasohlund — APPROVED** [review]

File: src/SqlPersistence/Outbox/OutboxPersister.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ class OutboxPersister : IOutboxStorage
 
     public OutboxPersister(IConnectionManager connectionManager, SqlDialect sqlDialect, OutboxCommands outboxCommands,
         Func<ISqlOutboxTransaction> outboxTransactionFactory,
-        int cleanupBatchSize = 4000)
+        int cleanupBatchSize = 4000) // Keep below 4000 to prevent lock escalation
```
Comment (andreasohlund) [inline src/SqlPersistence/Outbox/OutboxPersister.cs:23]: Nothing seems to be using this options so perhaps remove it and hardcode?

**Review by andreasohlund — APPROVED** [review]

**Comment by udidahan** [comment]
Given the approval, it seems like this was validated as having some benefit, so why was it not followed through on?

**Comment by ramonsmits** [comment]
> Given the approval, it seems like this was validated as having some benefit, so why was it not followed through on?

No specific reason except this didn't show up in the github notifications and due to your comment it did.

**Comment by udidahan** [comment]
I would have thought that the ping from the bot would have triggered the same notification, but ok.

**Pull request merged**